### ISSUE TITLE
computeInstanceMetrics - panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by cjwong1098** [qonto/prometheus-rds-exporter#10]
I am running 0.2.3 app version with aws role that has admin access
When I run this app locally against Account A (with only 1 db instance) it worked
However when I run this app locally against Account B (this one has multiple instances) I get the below error

```
{"time":"2023-10-12T16:13:00.840914676Z","level":"INFO","msg":"starting the HTTP server component"}
{"time":"2023-10-12T16:13:06.138535387Z","level":"DEBUG","msg":"received query"}
{"time":"2023-10-12T16:13:06.13861547Z","level":"INFO","msg":"get RDS metrics"}
{"time":"2023-10-12T16:13:06.138727887Z","level":"DEBUG","msg":"fetch quotas"}
{"time":"2023-10-12T16:13:06.13939422Z","level":"DEBUG","msg":"fetch usage metrics"}
{"time":"2023-10-12T16:13:06.311188054Z","level":"DEBUG","msg":"usage metrics fetched","metrics":{"AllocatedStorage":192824,"DBInstances":349,"ManualSnapshots":238,"ReservedDBInstances":0}}
{"time":"2023-10-12T16:13:06.589292012Z","level":"DEBUG","msg":"quota metrics fetched","metrics":{"DBinstances":1200,"TotalStorage":214748364800000,"ManualDBInstanceSnapshots":400}}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14b1df8]

goroutine 52 [running]:
github.com/qonto/prometheus-rds-exporter/internal/app/rds.(*RDSFetcher).computeInstanceMetrics(_, {0x0, 0x0, 0x0, {0x0, 0x0}, {0x0, 0x0}, {0x400043c028, 0x7}, ...}, ...)
	/home/runner/work/prometheus-rds-exporter/prometheus-rds-exporter/internal/app/rds/rds.go:227 +0x308
github.com/qonto/prometheus-rds-exporter/internal/app/rds.(*RDSFetcher).GetInstancesMetrics(0x40004f7790)
	/home/runner/work/prometheus-rds-exporter/prometheus-rds-exporter/internal/app/rds/rds.go:163 +0x24c
github.com/qonto/prometheus-rds-exporter/internal/app/exporter.(*rdsCollector).fetchMetrics(0x4000181680)
	/home/runner/work/prometheus-rds-exporter/prometheus-rds-exporter/internal/app/exporter/exporter.go:275 +0x208
github.com/qonto/prometheus-rds-exporter/internal/app/exporter.(*rdsCollector).Collect(0x4000181680, 0x4000667f58?)
	/home/runner/work/prometheus-rds-exporter/prometheus-rds-exporter/internal/app/exporter/exporter.go:379 +0x144
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
	/home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.16.0/prometheus/registry.go:455 +0xd8
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather in goroutine 37
	/home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.16.0/prometheus/registry.go:466 +0x4c8
```

I suspect the problem is related to the nil information returned somewhere

**Comment by vmercierfr** [comment]
Sorry about that issue. It sounds like it fails on the return value for the Performance insight, I tried a fix on `add-pi-information`.

Is there any chance that you can try the `add-pi-information` branch:

```
git clone git@github.com:qonto/prometheus-rds-exporter.git
cd prometheus-rds-exporter
git checkout add-pi-information
go run . --debug
```

Also, can you share which engines (e.g. PostgreSQL, MySQL, ...) are used in this AWS account (we may have some specificities)?

Thanks,

**Comment by cedric-oscar** [comment]
That worked for me. Thank you so much!

Do you need me to provide anymore information?

**Comment by cjwong1098** [comment]
That worked for me. Thank you so much!!

do you need me to provide anymore information?

**Comment by vmercierfr** [comment]
No, we are fine.
Thanks for taking the time to test the pre-release.

I created the pull request to fix it at https://github.com/qonto/prometheus-rds-exporter/pull/12. Once approved, we'll release a new minor version.

**Comment by vmercierfr** [comment]
I just released the [0.2.5](https://github.com/qonto/prometheus-rds-exporter/releases/tag/0.2.5) that contains this fix.

Thanks for reporting the issue.
